wrappers/nodejs/src/addon.cpp

```diff
@@ -1455,47 +1455,89 @@ Nan::Persistent<v8::Function> RSSyncer::constructor_;
 
 class Options {
  public:
-  explicit Options(rs2_options* options) : options_(options), error_(nullptr) {}
+  explicit Options(rs2_options* options = nullptr) : options_(options),
+      error_(nullptr) {}
 
-  ~Options() {
+  virtual ~Options() {
     if (error_) rs2_free_error(error_);
   }
 
-  bool SupportsOption(rs2_option option) {
-    return (rs2_supports_option(options_, option, &error_) != 0);
+  void SetOptionsPointer(rs2_options* options) {
+    options_ = options;
   }
 
-  float GetOption(rs2_option option) {
-    return rs2_get_option(options_, option, &error_);
+  void SupportsOptionInternal(
+      const Nan::FunctionCallbackInfo<v8::Value>& info) {
+    int32_t option = info[0]->IntegerValue();
+    auto on = rs2_supports_option(
+        options_, static_cast<rs2_option>(option), &error_);
+    info.GetReturnValue().Set(Nan::New(on ? true : false));
+    return;
   }
 
-  const char* GetOptionDescription(rs2_option option) {
-    return rs2_get_option_description(options_, option, &error_);
+  void GetOptionInternal(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+    int32_t option = info[0]->IntegerValue();
+    auto value = rs2_get_option(
+        options_, static_cast<rs2_option>(option), &error_);
+    info.GetReturnValue().Set(Nan::New(value));
   }
 
-  const char* GetOptionValueDescription(rs2_option option, float val) {
-    return rs2_get_option_value_description(options_, option, val, &error_);
+  void GetOptionDescriptionInternal(
+      const Nan::FunctionCallbackInfo<v8::Value>& info) {
+    int32_t option = info[0]->IntegerValue();
+    auto desc = rs2_get_option_description(
+        options_, static_cast<rs2_option>(option), &error_);
+    if (desc)
+      info.GetReturnValue().Set(Nan::New(desc).ToLocalChecked());
+    else
+      info.GetReturnValue().Set(Nan::Undefined());
   }
 
-  void SetOption(rs2_option option, float val) {
-    rs2_set_option(options_, option, val, &error_);
+  void GetOptionValueDescriptionInternal(
+      const Nan::FunctionCallbackInfo<v8::Value>& info) {
+    int32_t option = info[0]->IntegerValue();
+    auto val = info[1]->NumberValue();
+    auto desc = rs2_get_option_value_description(
+        options_, static_cast<rs2_option>(option), val, &error_);
+    if (desc)
+      info.GetReturnValue().Set(Nan::New(desc).ToLocalChecked());
+    else
+      info.GetReturnValue().Set(Nan::Undefined());
```

Comment:
Use?:

Reply:
As the Nan::New(desc)... and Nan::Undefined() are of different types, if we use ? : we will face compile issue. And if we cast them into one type, we will create a very long statement, so I suggest to keep it separated which may be clear to read.

Reply:
oh, I remember, so keep it as now.